File: medium/1288-remove-covered-intervals/solution.java
import java.util.Arrays;

/**
 * 1288. Remove Covered Intervals
 * Time: O(n log n)
 * Space: O(1)
 */
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        // left ascending, right descending on ties (bigger interval first)
        Arrays.sort(intervals, (a, b) -> a[0] != b[0] ? a[0] - b[0] : b[1] - a[1]);
        int count = 0;
        int maxRight = -1;
        for (int[] iv : intervals) {
            if (iv[1] > maxRight) { // not covered by any previous interval
                count++;
                maxRight = iv[1];
            }
        }
        return count;
    }
}

File: medium/1288-remove-covered-intervals/solution.py
from typing import List


class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        """
        1288. Remove Covered Intervals
        Time: O(n log n)
        Space: O(1)
        """
        # left ascending, right descending on ties (bigger interval first)
        intervals.sort(key=lambda x: (x[0], -x[1]))
        count = 0
        max_right = -1
        for _, r in intervals:
            if r > max_right:      # not covered by any previous interval
                count += 1
                max_right = r
        return count
